assets/js/common/DateFilter/DateFilter.jsx

```diff
@@ -0,0 +1,213 @@
+import React, { Fragment, useState, useRef } from 'react';
+import classNames from 'classnames';
+import { Transition } from '@headlessui/react';
+
+import useOnClickOutside from '@hooks/useOnClickOutside';
+import { EOS_CLOSE } from 'eos-icons-react';
+
+
+const oneHour = 60 * 60 * 1000;
+const preconfiguredOptions = {
+  '1h ago': () => new Date(Date.now() - oneHour),
+  '24h ago': () => new Date(Date.now() - 24 * oneHour),
+  '7d ago': () => new Date(Date.now() - 7 * 24 * oneHour),
+  '30d ago': () => new Date(Date.now() - 30 * 24 * oneHour),
+};
+
+const renderOptionItem = (option, placeholder) => {
+  if (!option || !Array.isArray(option)) {
+    return placeholder;
+  }
+  if (typeof option[2] === 'function') {
+    return option[2]();
+  }
+  return option[0];
+};
+
+
+const parseInputOptions = (options) =>
+  options
+    .map((option) => {
+      if (typeof option === 'string' && option in preconfiguredOptions) {
+        return [option, preconfiguredOptions[option]];
+      }
+      if (
+        Array.isArray(option) &&
+        typeof option[1] === 'function' &&
+        option[1]() instanceof Date
+      ) {
+        return option;
+      }
+      return undefined;
+    })
+    .filter(Boolean)
+    .reduceRight(
+      (acc, el) =>
+        acc.find(([label]) => label === el[0]) ? acc : [...acc, el],
+      []
+    )
+    .sort((a, b) => b[1]().getTime() - a[1]().getTime());
+
+const getSelectedOption = (options, value) => {
+  const selectedId = Array.isArray(value) ? value[0] : value;
```

Comment:
**decision** Ideally, the option ID is enough to set the selected item. However, the value exposed on change is a couple `[ID, date]` and I thought it would be convenient to accept it as value as well otherwise it would require us to handle the difference in format when storing it in the local/global state.